features/26-mfa/api.md

```diff
@@ -0,0 +1,436 @@
+
+  const code = textInput.value;
+  const { bearerToken } = await skygear.auth.mfa.authenticateWithOOB({ authenticatorID, code }, { requestBearerToken: true});
+
+  // Persist the bearer token
+  if (bearerToken) {
+    await storeBearerToken(bearerToken);
+  }
+}
+```
+
+The above example demonstrates the flow of authenitcation in details.
+The bearer token should be handled transparently by the SDK so the actual code written by the developer should not need to be aware of bearer token.
+
+## List authenticators in settings screen
+
+```typescript
+const authenticators = await skygear.auth.mfa.listAuthenticators();
+// [
+//   {
+//     id: "1",
+//     type: "totp",
+//     activatedAt: new Date("2019-07-19T00:00:00.000Z"),
+//     displayName: "totp-2edfdca2-3f05-41d2-bfd8-502725ee09a6",
+//   },
+//   {
+//     id: "2",
+//     type: "oob",
+//     activatedAt: new Date("2019-07-19T00:00:00.000Z"),
+//     channel: "sms",
+//     phone: "+85223456789",
+//   },
+//   {
+//     id: "3",
+//     type: "recovery_code",
+//     createdAt: new Date("2019-07-19T00:00:00.000Z"),
+//   },
+//   {
+//     id: "4",
+//     type: "recovery_code",
+//     createdAt: new Date("2019-07-19T00:00:00.000Z"),
+//   },
+//   {
+//     id: "5",
+//     type: "recovery_code",
+//     createdAt: new Date("2019-07-19T00:00:00.000Z"),
+//   },
+//   {
+//     id: "5",
+//     type: "recovery_code",
+//     createdAt: new Date("2019-07-19T00:00:00.000Z"),
+//   },
+//   {
+//     id: "6",
+//     type: "bearer_token",
+//     createdAt: new Date("2019-07-19T00:00:00.000Z"),
+//     expireAt: new Date("2019-08-19T00:00:00.000Z"),
+//   }
+// ]
+
+const numberOfRecoveryCodes = authenticators.filter(a => a.type === "recovery_code").length;
+const hasTOTP = authenticators.filter(a => a.type === "totp").length > 0;
+const hasOOB = authenticators.filter(a => a.type === "oob").length > 0;
+```
+
+## Delete an authenticator
+
+```typescript
+const authenticators = await skygear.auth.mfa.listAuthenticators();
+// Present an UI to show the list of authenticators and
+// let the user to choose which one to delete.
+
+// Suppose the user wants to delete the first one.
+const { id } = authenticators[0];
+await skygear.auth.mfa.deleteAuthenticator(id);
+
+// The user can also delete a bearer token if they believe
+// the device storing it has been compromised.
+const bearerTokens = authenticators.filter(a => a.type === "bearer_token");
+const { id } = bearerTokens[0];
+await skygear.auth.mfa.deleteAuthenticator(id);
+```
+
+## Regenerate recovery codes
+
+```typescript
+const { recoveryCodes } = await skygear.auth.mfa.regenerateRecoveryCode();
+// Present an UI to show the list of recovery codes to the user.
+```
+
+# Access Token
+
+The access token includes a new claim [amr](https://openid.net/specs/openid-connect-core-1_0.html).
+
+## Mapping between existing authentication method and factor name
+
+|Method|Factor Name|
```

Comment:
There exists a list of defined factors: https://tools.ietf.org/html/rfc8176

Reply:
Thanks for letting me know! If the user is authenticated with password and OOB, then `amr` should be `["pwd", "mfa", "sms"]`